families/supplychain/python/sawtooth_supplychain/rest_api/sc_rest_api.py

```diff
@@ -0,0 +1,162 @@
+from sawtooth_supplychain.rest_api.config import RestApiConfig
+from sawtooth_supplychain.rest_api.route_handlers import RouteHandler
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+def parse_args(args):
+    """Parse command line flags added to `rest_api` command.
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-B', '--bind',
+                        help='The host and port for the api to run on.',
+                        action='append')
+    parser.add_argument('-C', '--connect',
+                        help='The url to connect to a running Validator')
+    parser.add_argument('-t', '--timeout',
+                        help='Seconds to wait for a validator response')
+    parser.add_argument('-v', '--verbose',
+                        action='count',
+                        default=0,
+                        help='Increase level of output sent to stderr')
+    parser.add_argument('-D', '--db_cnx',
+                        action='store',
+                        required=True,
+                        help='The database connection string')
+
+    return parser.parse_args(args)
+
+
+async def cors_handler(request):
+    headers = {}
+    RouteHandler.add_cors_headers(request, headers)
+    return web.Response(headers=headers)
+
+
+def start_rest_api(host, port, connection, timeout, db_cnx):
+    """Builds the web app, adds route handlers, and finally starts the app.
+    """
+    loop = asyncio.get_event_loop()
+    connection.open()
+    app = web.Application(loop=loop)
+    app.on_cleanup.append(lambda app: connection.close())  # new
+
+    # Add routes to the web app
+    LOGGER.info('Creating handlers for validator at %s', connection.url)
+
+    handler = RouteHandler(loop, connection, timeout, db_cnx)
+
+    app.router.add_route('OPTIONS', '/{route_name}', cors_handler)
+
+    app.router.add_get('/agents', handler.list_agents)
+    app.router.add_get('/agents/{agent_id}', handler.fetch_agent)
+    app.router.add_get('/applications', handler.list_applications)
+    app.router.add_get('/records', handler.list_records)
+    app.router.add_get('/records/{record_id}', handler.fetch_record)
+    app.router.add_get('/records/{record_id}/applications',
+                       handler.fetch_applications)
+
+    # Start app
+    LOGGER.info('Starting Supplychain REST API on "%s:%s', host, port)
+    web.run_app(app, host=host, port=port,
+                access_log=LOGGER,
+                access_log_format='%r: %s status, %b size, in %Tf s')
+
+
+def load_rest_api_config(first_config):
+    default_config = load_default_rest_api_config()
+    config_dir = get_config_dir()
+    conf_file = os.path.join(config_dir, 'rest_api.toml')
+
+    toml_config = load_toml_rest_api_config(conf_file)
+    return merge_rest_api_config(
+        configs=[first_config, toml_config, default_config])
+
+
+def main():
+    loop = ZMQEventLoop()
+    asyncio.set_event_loop(loop)
+
+    connection = None
```

Comment:
Nothing should be using a connection to the validator, currently, so there is no need to create a connection and pass it around.